src/acl/FilledChecklist.cc

```diff
@@ -263,3 +259,22 @@ ACLFilledChecklist::setIdent(const char *ident)
 #endif
 }
 
+void
+ACLFilledChecklist::updateAle(const AccessLogEntry::Pointer &a)
+{
+    if (!a)
+        return;
+
+    al = a; // could have been set already (to a different value)
+    if (!request)
+        setRequest(a->request);
+    updateReply(a->reply);
+}
+
+void
+ACLFilledChecklist::updateReply(const HttpReply::Pointer &r)
+{
+    if (r)
+        reply_ = r; // may already be set, including to r
```

Comment:
This breaks refcounting which already handles the nil check properly.
```suggestion
    reply_ = r; // refcounted
```